xchange-core/src/main/java/org/knowm/xchange/dto/account/Wallet.java

```diff
@@ -24,15 +20,24 @@
   private String id;
   /** A descriptive name for this wallet. Defaults to {@link #id} */
   private String name;
+  /** All wallet operations */
+  private final Set<WalletFeature> walletFeatures;
+  /** Maximum leverage that wallet supports */
+  private BigDecimal maxLeverage = BigDecimal.ZERO;
+  /** Current leverage of wallet */
+  private BigDecimal currentLeverage = BigDecimal.ZERO;
 
   /**
    * Constructs a {@link Wallet}.
    *
    * @param id the wallet id
    * @param name a descriptive name for the wallet
    * @param balances the balances, the currencies of the balances should not be duplicated.
+   * @param walletFeatures all the features that wallet supports
+   *     <p>maxLeverage and currentLeverage are BigDecimal.ZERO for the default constructor
    */
-  public Wallet(String id, String name, Collection<Balance> balances) {
+  public Wallet(
```

Comment:
these constructors are becoming a mess
maybe we should add a Builder class and deprecate them all?

Reply:
Totally agree